Feat/create crud for waste category

Figaarillo opened the pull request.

Figaarillo committed:
feat: create waste category payload

Figaarillo committed:
feat: create waste category entity

Figaarillo committed:
feat: create waste category repository

Figaarillo committed:
feat: create use case to register new waste category

Figaarillo committed:
feat: add implementation of waste category

Figaarillo committed:
feat: add zod schema validator

Figaarillo committed:
feat: create dto types

Figaarillo committed:
feat: create dto to check id

Figaarillo committed:
feat: create dto to register and update a waste category

Figaarillo committed:
feat: add waste category handler

Figaarillo committed:
feat: add waste category routes

Figaarillo committed:
feat: add bootstrap for waste category

Figaarillo committed:
feat: create use case to list all categories

Figaarillo committed:
feat: create use case to list find category by id

Figaarillo committed:
feat: create domain errors

Figaarillo committed:
feat: create use case to update waste category

Figaarillo committed:
feat: create use case to delete waste category

Figaarillo committed:
chore: modify path of routes for waste category

Figaarillo committed:
chore: rename function to bootstrap waste category

Figaarillo committed:
chore: register waste category in app

Figaarillo committed:
feat: add migrations

Figaarillo committed:
fix: remove extra attributes from category payload

wink15 commented on src/waste-category/domain/repositories/waste-category.repository.ts:
save: (category: WasteCategoryEntity) => Promise<Nullable<WasteCategoryEntity>> aca no se estaria updateando solamente la descripcion? se deberia poder updatear el nombre

Figaarillo replied:
No lo veo mal. Es una categoría y creo que se podría cambiar el nombre que se le asigna.

Figaarillo replied:
Ya agregue que se pueda actualizar el nombre de una categoría. Revísalo por si hay algún error, y si no hay, aprobá el PR.

wink15 reviewed (approved):
Comente eso del update para ver si se puede tambien updatear el nombre

Figaarillo committed:
refactor: modify the attributes that will be updated

The payload received to update a category was modified to accept the
name attribute.
